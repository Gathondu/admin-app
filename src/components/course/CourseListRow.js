import React from 'react';
import PropTypes from 'prop-types';
import {Link} from "react-router-dom";

const CourseListRow = ({course, deleteCourse}) => {
    return (
        <tr>
            <td><i className="fa fa-trash" onClick={() => deleteCourse(course.id)}/></td>
            <td><a href={course.watchHref} target="_blank" rel="noopener noreferrer"><i className="fa fa-television"/></a></td>
            <td><Link to={`/course/${course.id}`}>{course.title}</Link></td>
            <td>{course.authorId}</td>
            <td>{course.category}</td>
            <td>{course.length}</td>
        </tr>
    );
};

CourseListRow.propTypes = {
    course: PropTypes.object.isRequired
};

export default CourseListRow;
