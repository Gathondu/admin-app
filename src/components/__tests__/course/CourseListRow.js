import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import CourseListRow from "../../course/CourseListRow";

describe('CourseListRow Component', () => {
    const setup = () => {
        const props = {
            course: {},
            deleteCourse: jest.fn()
        };
        return {
            props,
            wrapper: shallow(<CourseListRow {...props}/>)
        }
    };
    it('does not regress', ()=>{
        const {wrapper} = setup();
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should delete course correctly', () => {
        const {wrapper, props} = setup();

        wrapper.find('.fa-trash').simulate('click');
        expect(props.deleteCourse).toHaveBeenCalled()
    });
});