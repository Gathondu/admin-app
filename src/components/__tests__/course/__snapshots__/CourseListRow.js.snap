// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseListRow Component does not regress 1`] = `
<tr>
  <td>
    <i
      className="fa fa-trash"
      onClick={[Function]}
    />
  </td>
  <td>
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fa fa-television"
      />
    </a>
  </td>
  <td>
    <Link
      replace={false}
      to="/course/undefined"
    />
  </td>
  <td />
  <td />
  <td />
</tr>
`;
